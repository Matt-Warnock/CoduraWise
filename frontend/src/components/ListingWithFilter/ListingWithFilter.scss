.container {
  display: flex;
  justify-content: space-around;

  & .resource-list {
    width: 100%;
  }
}
