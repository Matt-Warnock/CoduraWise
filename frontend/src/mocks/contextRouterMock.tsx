import ResourcesContextProvider from "../store/ResourcesContext";
import React from "react";
import { MemoryRouter } from "react-router";
import ToastsContextProvider from "../store/ToastsContext";

export default function ContextRouterMock({
  children,
  initialEntries,
}: {
  children: JSX.Element;
  initialEntries?: Array<string>;
}) {
  return (
    <ToastsContextProvider>
      <ResourcesContextProvider>
        <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
      </ResourcesContextProvider>
    </ToastsContextProvider>
  );
}
