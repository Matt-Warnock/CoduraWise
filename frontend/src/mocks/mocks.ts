import { Resource } from "../models/Resource";

import { Tag } from "../models/Tags";

export const mockResources: Array<Resource> = [
  { id: 1, title: "Java", link: "Java Link", mediaType: "video", averageRating: 1 },
  { id: 2, title: "Python", link: "Python Link", mediaType: "article", averageRating: 4 },
  { id: 3, title: "Javascript", link: "Javascript Link", mediaType: "course", averageRating: 3 },
  { id: 4, title: "C#", link: "C# Link", mediaType: "article", averageRating: 4 },
];

export const mockTags: Array<Tag> = [
  { tag: "Java" },
  { tag: "JavaScript" },
  { tag: "Agile" },
  { tag: "React" },
  { tag: "Beginners" },
  { tag: "HTML" },
  { tag: "Leadership" },
  { tag: "S1" },
  { tag: "S2" },
  { tag: "S3" },
];
