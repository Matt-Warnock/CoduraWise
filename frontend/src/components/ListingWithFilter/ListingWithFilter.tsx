import React, { useState } from "react";
import { MediaType } from "../../models/MediaTypes";
import { Resource } from "../../models/Resource";
import MediaTypeSelection from "../MediaTypeSelection/MediaTypeSelection";
import ResourceList from "../ResourceList/ResourceList";
import "./ListingWithFilter.scss";

interface ListingWithFilterProps {
  resources: Array<Resource>;
}

const ListingWithFilter = ({ resources }: ListingWithFilterProps) => {
  const [filterMediaTypes, setFilterMediaTypes] = useState<Array<MediaType>>(
    [],
  );

  return (
    <div className="container">
      <div className="resource-list">
        <ResourceList
          resources={resources}
          filterMediaTypes={filterMediaTypes}
        />
      </div>
      <MediaTypeSelection setFilterMediaTypes={setFilterMediaTypes} />
    </div>
  );
};

export default ListingWithFilter;
