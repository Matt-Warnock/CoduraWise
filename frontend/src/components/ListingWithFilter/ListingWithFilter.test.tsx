import TestRenderer from "react-test-renderer";
import { mockResources } from "../../mocks/mocks";
import ListingWithFilter from "./ListingWithFilter";
import React from "react";
import ContextRouterMock from "../../mocks/contextRouterMock";

test("snapshot test: the html matches", () => {
  const listingWithFilterComponent = TestRenderer.create(
    <ContextRouterMock>
      <ListingWithFilter resources={mockResources} />
    </ContextRouterMock>
  );

  expect(listingWithFilterComponent.toJSON()).toMatchSnapshot();
});
