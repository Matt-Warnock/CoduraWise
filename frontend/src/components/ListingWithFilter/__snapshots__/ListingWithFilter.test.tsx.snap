// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: the html matches 1`] = `
<div
  className="container"
>
  <div
    className="resource-list"
  >
    <ul
      className="resource-list__container"
    >
      <li
        className="resource-list__list-element"
      >
        <article
          className="resource-card__container"
        >
          <h3
            className="resource-card__title"
          >
            <a
              className="resource-card__link"
              href="Java Link"
              rel="noreferrer"
              target="_blank"
            >
              Java
            </a>
          </h3>
          <div
            aria-label="Rating of this product is 1 out of 5."
            className="star-container"
            style={
              Object {
                "--rating": 1,
              }
            }
          />
        </article>
      </li>
      <li
        className="resource-list__list-element"
      >
        <article
          className="resource-card__container"
        >
          <h3
            className="resource-card__title"
          >
            <a
              className="resource-card__link"
              href="Python Link"
              rel="noreferrer"
              target="_blank"
            >
              Python
            </a>
          </h3>
          <div
            aria-label="Rating of this product is 4 out of 5."
            className="star-container"
            style={
              Object {
                "--rating": 4,
              }
            }
          />
        </article>
      </li>
      <li
        className="resource-list__list-element"
      >
        <article
          className="resource-card__container"
        >
          <h3
            className="resource-card__title"
          >
            <a
              className="resource-card__link"
              href="Javascript Link"
              rel="noreferrer"
              target="_blank"
            >
              Javascript
            </a>
          </h3>
          <div
            aria-label="Rating of this product is 3 out of 5."
            className="star-container"
            style={
              Object {
                "--rating": 3,
              }
            }
          />
        </article>
      </li>
      <li
        className="resource-list__list-element"
      >
        <article
          className="resource-card__container"
        >
          <h3
            className="resource-card__title"
          >
            <a
              className="resource-card__link"
              href="C# Link"
              rel="noreferrer"
              target="_blank"
            >
              C#
            </a>
          </h3>
          <div
            aria-label="Rating of this product is 4 out of 5."
            className="star-container"
            style={
              Object {
                "--rating": 4,
              }
            }
          />
        </article>
      </li>
    </ul>
  </div>
  <form
    className="media-selection"
  >
    <legend
      className="media-selection__legend"
    >
      Filter by
    </legend>
    <div>
      <input
        checked={false}
        className="media-selection__input"
        id="videomedia"
        name="video"
        onChange={[Function]}
        type="checkbox"
        value="video"
      />
      <label
        className="media-selection__label"
        htmlFor="videomedia"
      >
        video
      </label>
    </div>
    <div>
      <input
        checked={false}
        className="media-selection__input"
        id="articlemedia"
        name="article"
        onChange={[Function]}
        type="checkbox"
        value="article"
      />
      <label
        className="media-selection__label"
        htmlFor="articlemedia"
      >
        article
      </label>
    </div>
    <div>
      <input
        checked={false}
        className="media-selection__input"
        id="bookmedia"
        name="book"
        onChange={[Function]}
        type="checkbox"
        value="book"
      />
      <label
        className="media-selection__label"
        htmlFor="bookmedia"
      >
        book
      </label>
    </div>
    <div>
      <input
        checked={false}
        className="media-selection__input"
        id="tutorialmedia"
        name="tutorial"
        onChange={[Function]}
        type="checkbox"
        value="tutorial"
      />
      <label
        className="media-selection__label"
        htmlFor="tutorialmedia"
      >
        tutorial
      </label>
    </div>
    <div>
      <input
        checked={false}
        className="media-selection__input"
        id="coursemedia"
        name="course"
        onChange={[Function]}
        type="checkbox"
        value="course"
      />
      <label
        className="media-selection__label"
        htmlFor="coursemedia"
      >
        course
      </label>
    </div>
  </form>
</div>
`;
